### DISCUSSION TITLE
Check if a user has a password set

### DISCUSSION THREAD

**Discussion by baderj** [pocketbase/pocketbase#5754] [Q&A] [answered]
My applications offers both social logins and logins by email and password. Now depending on whether a user has a password set, I would like to hide or show the password reset functionality. 

There is the `listExternalAuths(<userid>)` method to list which external auths are linked to a particular user, but how can I figure out whether a user has a password set or not?

**Comment by ganigeorgiev** [comment, marked as answer]
There is no support for this out of the box because all users have a password even if not explicitly set (_when the user is created via OAuth2 the password is autogenerated; there are even plans to eventually return the autogenerated password so that it can be treated as "security code" but remains in my todo for later consideration_).

Based on my personal experience, in this type of situations I usually either
- don't show "old+new password form" to the user and instead if the user has an email simply have a "Send password change request" link that calls `requestPasswordReset(pb.authStore.model.email)`.
- have a bool "hasPassword" or similar flag that is automatically set either via the hooks or manually as create/update call

**Reply by baderj** [reply]
Thanks a lot, this makes perfect sense. I didn't expect the password reset to work if none was set in the first place by the user, but since the reset works regardless I think I'll just have a the reset link shown to all users.
